auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java

```diff
@@ -140,7 +140,8 @@ public void onClick(View view) {
                     }
                 });
                 provider.setAuthenticationCallback(this);
-                btnHolder.addView(loginButton, 0);
+                // Account for the email provider which should always be at the bottom
+                btnHolder.addView(loginButton, btnHolder.getChildCount() - 1);
```

Comment:
If we're going to have a deprecated version of the old behavior anyway then I agree, let's just fully respect the order they provide.